### PR TITLE
`newrelic` - upgrade API version to `2024-03-01`

### PR DESCRIPTION
**Pull request by teowa**
## Community Note

* Please vote on this PR by adding a :thumbsup: [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original PR to help the community and maintainers prioritize for review
* Please do not leave comments along the lines of "+1", "me too" or "any updates", they generate extra noise for PR followers and do not help prioritize for review

## PR Checklist

- [x] I have followed the guidelines in our [Contributing Documentation](../blob/main/contributing/README.md).
- [x] I have checked to ensure there aren't other open [Pull Requests](../pulls) for the same update/change.
- [x] I have checked if my changes close any open issues. If so please include appropriate [closing keywords](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword) below.
- [x] I have updated/added Documentation as required written in a helpful and kind way to assist users that may be unfamiliar with the resource / data source.
- [x] I have used a meaningful PR title to help maintainers and other users understand this change and help prevent duplicate work. 
For example: “`resource_name_here` - description of change e.g. adding property `new_property_name_here`”

## Changes to existing Resource / Data Source

- [x] I have added an explanation of what my changes do and why I'd like you to include them (This may be covered by linking to an issue above, but may benefit from additional explanation).
- [x] I have written new tests for my resource or datasource changes & updated any relevent documentation.
- [x] I have successfully run tests with my changes locally. If not, please provide details on testing challenges that prevented you running the tests.
- [ ] (For changes that include a **state migration only**). I have manually tested the migration path between relevant versions of the provider.

## Testing 

- [x] My submission includes Test coverage as described in the [Contribution Guide](../blob/main/contributing/topics/guide-new-resource.md) and the tests pass. (if this is not possible for any reason, please include details of why you did or could not add test coverage)

## Change Log

Below please provide what should go into the changelog (if anything) conforming to the [Changelog Format documented here](../blob/main/contributing/topics/maintainer-changelog.md).

* `newrelic` - upgrade API version to `2024-03-01`

This is a (please select all that apply):

- [ ] Bug Fix
- [ ] New Feature (ie adding a service, resource, or data source)
- [x] Enhancement
- [ ] Breaking Change

## Related Issue(s)
Fixes #0000

> [!NOTE] 
> If this PR changes meaningfully during the course of review please update the title and description as required.

### CODE REVIEW

**Comment by teowa** [comment]
image

File: internal/services/newrelic/new_relic_monitor_resource.go [hunk 1]
```diff
@@ -105,11 +105,11 @@ func (r NewRelicMonitorResource) Arguments() map[string]*pluginsdk.Schema {
...
 						ForceNew: true,
-						Default:  string(monitors.BillingCycleMONTHLY),
+						Default:  "MONTHLY",
 						ValidateFunc: validation.StringInSlice([]string{
-							string(monitors.BillingCycleMONTHLY),
-							string(monitors.BillingCycleWEEKLY),
-							string(monitors.BillingCycleYEARLY),
+							"MONTHLY",
+							"WEEKLY",
+							"YEARLY",
```
Comment (katbyte) [inline internal/services/newrelic/new_relic_monitor_resource.go:112]: is there a reason these are no longer constants? could we open an issue on the rest specs asking and link it here ?

**Review by katbyte — CHANGES_REQUESTED** [review]

File: internal/services/newrelic/new_relic_monitor_resource.go (hunk 1 above)
Comment (teowa) [reply internal/services/newrelic/new_relic_monitor_resource.go:112]: Hi @katbyte , from service team, earlier there was limit on values that this field can have, now that limitation has lifted. There will be new values in the future.

**Comment by teowa** [comment]
Why the Enum is changed to string type in new API version:

From service team, earlier there was limit on values that this field can have, now that limitation has lifted. There will be new values in the future.

File: internal/services/newrelic/new_relic_monitor_resource.go (hunk 1 above)
Comment (katbyte) [reply internal/services/newrelic/new_relic_monitor_resource.go:112]: I'm not sure why that means there is no longer an enum/defined in the rest API specs? i cannot see any reason why these values shouldn't be captures there. could we place open an issue on the rest specs + link it here?

**Comment by katbyte** [comment]
we have test failures:
```
------- Stdout: -------
=== RUN   TestAccNewRelicTagRule_complete
=== PAUSE TestAccNewRelicTagRule_complete
=== CONT  TestAccNewRelicTagRule_complete
    testcase.go:173: Step 1/3 error: Error running apply: exit status 1
        
        Error: creating Monitor (Subscription: "*******"
        Resource Group Name: "acctest-rg-240924155651927473"
        Monitor Name: "acctest-nrm-240924155651927473"): performing CreateOrUpdate: unexpected status 400 (400 Bad Request) with error: BadRequest: Validation Failed: Managed Identity URI not found
        
          with azurerm_new_relic_monitor.test,
          on terraform_plugin_test.tf line 36, in resource "azurerm_new_relic_monitor" "test":
          36: resource "azurerm_new_relic_monitor" "test" {
        
        creating Monitor (Subscription: "*******"
        Resource Group Name: "acctest-rg-240924155651927473"
        Monitor Name: "acctest-nrm-240924155651927473"): performing CreateOrUpdate:
        unexpected status 400 (400 Bad Request) with error: BadRequest: Validation
        Failed: Managed Identity URI not found
--- FAIL: TestAccNewRelicTagRule_complete (127.37s)
FAIL
```

and could we open a rest API specs issue calling out the change away from an enum (i see no reason why it shouldn't be one) and link it in the code. once that is done + tests pass this should be good to merge

**Comment by teowa** [comment]
image

**Comment by katbyte** [comment]
@teowa - could we please address this last comment? https://github.com/hashicorp/terraform-provider-azurerm/pull/27135#discussion_r1773677096 and link a rest specs issue? that property should still be an enum

**Review by katbyte — APPROVED** [review]
Thanks! LGTM now 🚜

**Pull request merged**